src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1971,16 +1972,45 @@ ExecuteRebalancerCommandInSeparateTransaction(char *command)
 	MultiConnection *connection = GetNodeConnection(connectionFlag, LocalHostName,
 													PostPortNumber);
 	StringInfo setApplicationName = makeStringInfo();
-	appendStringInfo(setApplicationName, "SET application_name TO %s",
+	appendStringInfo(setApplicationName, "SET application_name TO %s;",
 					 CITUS_REBALANCER_NAME);
 
+	StringInfo setStatements = GetSetStatementsForNewConnections();
+	appendStringInfoString(setApplicationName, setStatements->data);
+
 	ExecuteCriticalRemoteCommand(connection, setApplicationName->data);
 	ExecuteCriticalRemoteCommand(connection, command);
 
 	CloseConnection(connection);
 }
 
 
+StringInfo
+GetSetStatementsForNewConnections(void)
+{
+	StringInfo setStatements = makeStringInfo();
+
+	List *variableList = NIL;
+	char *splitCopy = pstrdup(VariablesToBePassedToNewConnections);
+
+	if (!SplitGUCList(splitCopy, ',', &variableList))
+	{
+		GUC_check_errdetail("not a valid list of identifiers");
+		return false;
+	}
+
+	char *variableName = NULL;
+	foreach_ptr(variableName, variableList)
+	{
+		const char *variableValue = GetConfigOption(variableName, true, true);
+		appendStringInfo(setStatements, "SET %s TO %s;",
```

Comment:
If pg_dist_poolinfo is set up with a transaction based pooler. These SET commands will be lost. To work around this we could make ExecuteRebalancerCommandInSeparateTransaction open a transaction instead:

BEGIN;
SET LOCAL ...
SET LOCAL ...
<command>
COMMIT;